flake.nix

```diff
@@ -13,77 +12,89 @@
   outputs = {
     self,
     nixpkgs,
-    flake-utils,
     rust-overlay,
     ...
   }: let
-    gitRev = self.rev or self.dirtyRev or null;
-  in
-    flake-utils.lib.eachDefaultSystem (system: let
-      pkgs = import nixpkgs {
-        inherit system;
+    inherit (nixpkgs) lib;
+    systems = [
+      "x86_64-linux"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ];
+    eachSystem = lib.genAttrs systems;
+    pkgsFor = eachSystem (system:
+      import nixpkgs {
+        localSystem.system = system;
         overlays = [(import rust-overlay)];
```

Comment:
Does this line need to include the overlay from below? Looks like `self.packages.${system}.helix` is pointing to the nixpkgs `helix`:

```sh
$ nix run . -- --version
helix 25.01.1 (e7ac2fcd)
```